google-cloud-firestore/src/main/java/com/google/cloud/firestore/UpdateBuilder.java

```diff
@@ -595,22 +585,11 @@ private T performDelete(
     request.setDatabase(firestore.getDatabaseName());
 
     for (Mutation mutation : mutations) {
-      Preconditions.checkState(
-          mutation.document != null || mutation.transform != null,
-          "Either a write or transform must be set");
-
       if (mutation.precondition != null) {
-        (mutation.document != null ? mutation.document : mutation.transform)
-            .setCurrentDocument(mutation.precondition);
-      }
-
-      if (mutation.document != null) {
-        request.addWrites(mutation.document);
+        mutation.document.setCurrentDocument(mutation.precondition);
```

Comment:
If we use the Write proto directly instead of `Mutation`, we should be able to get rid of this line.

Reply:
Yep, sorry - I thought this was in "commit()". Looks good.